### ISSUE TITLE
RoadRunner does not use default steps=50

### ISSUE THREAD

**Issue by matthiaskoenig** [sys-bio/roadrunner#249]
Somehow roadrunner is not integrating models when steps are not set via simulate (despite having a default value of 50 steps).

Example code
```
%matplotlib inline

# Simulating simple test model
import roadrunner

# from simapp.models import CompModel
# mtest = CompModel.objects.get(model_id="test_3")
# print(mtest)
# r = roadrunner.RoadRunner(mtest.filepath)
r = roadrunner.RoadRunner("test_3.xml")

# settings
absTol = 1E-6
relTol = 1E-6
absTol = absTol * min(r.model.getCompartmentVolumes())  # absTol relative to the amounts
integrator = r.getIntegrator()
integrator.setValue('stiff', True)
integrator.setValue('absolute_tolerance', absTol)
integrator.setValue('relative_tolerance', relTol)

# integrate
r.simulate(0,10)
print(r)
print('-'*80)
integrator = r.getIntegrator()
print(integrator.getName())
for key in integrator.getSettings():
    print(key, ':', integrator.getValue(key))
# plot
r.plot()

# => does not simulate, despite the steps default setting of 50 ????

# setting the steps explcitely is working
# r.simulate(0,10, 100)
# r.simulate(0,10, steps=100)
```
Example model
```
<?xml version="1.0" encoding="UTF-8"?>
<!-- Created by multiscale-modelcreator version 0.1.0 on 2016-01-11 15:13 with libSBML version 5.12.1. -->
<sbml xmlns="http://www.sbml.org/sbml/level3/version1/core" xmlns:fbc="http://www.sbml.org/sbml/level3/version1/fbc/version2" level="3" version="1" fbc:required="false">
  <model metaid="meta_8bb240b7a6de47339464002234653772" id="test_3" name="test_3" substanceUnits="mole" timeUnits="s" volumeUnits="m3" areaUnits="m2" lengthUnits="m" extentUnits="mole" fbc:strict="false">
    <notes>
      <body xmlns="http://www.w3.org/1999/xhtml">
        <h1>Koenig Test Model</h1>
        <h2>Description</h2>
        <h2>Terms of use</h2>
        <div class="dc:rightsHolder">Copyright © 2015 Matthias Koenig.</div>
        <div class="dc:license">
          <p>Redistribution and use of any part of this model, with or without modification, are permitted provided that the following conditions are met:
        <ol>
            <li>Redistributions of this SBML file must retain the above copyright notice, this list of conditions and the following disclaimer.</li>
            <li>Redistributions in a different form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided
          with the distribution.</li>
          </ol>
        This model is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
        </p>
      </div>
    </body>
  </notes>
  <annotation>
    <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dcterms="http://purl.org/dc/terms/" xmlns:vCard="http://www.w3.org/2001/vcard-rdf/3.0#" xmlns:bqbiol="http://biomodels.net/biology-qualifiers/" xmlns:bqmodel="http://biomodels.net/model-qualifiers/">
      <rdf:Description rdf:about="#meta_8bb240b7a6de47339464002234653772">
        <dc:creator>
          <rdf:Bag>
            <rdf:li rdf:parseType="Resource">
              <vCard:N rdf:parseType="Resource">
                <vCard:Family>Koenig</vCard:Family>
                <vCard:Given>Matthias</vCard:Given>
              </vCard:N>
              <vCard:EMAIL>konigmatt@googlemail.com</vCard:EMAIL>
              <vCard:ORG rdf:parseType="Resource">
                <vCard:Orgname>Charite Berlin</vCard:Orgname>
              </vCard:ORG>
            </rdf:li>
          </rdf:Bag>
        </dc:creator>
        <dcterms:created rdf:parseType="Resource">
          <dcterms:W3CDTF>2016-01-11T15:13:07Z</dcterms:W3CDTF>
        </dcterms:created>
        <dcterms:modified rdf:parseType="Resource">
          <dcterms:W3CDTF>2016-01-11T15:13:07Z</dcterms:W3CDTF>
        </dcterms:modified>
      </rdf:Description>
    </rdf:RDF>
  </annotation>
  <listOfUnitDefinitions>
    <unitDefinition id="Pa_s_per_m2">
      <listOfUnits>
        <unit kind="pascal" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="-2" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="Pa_s_per_m3">
      <listOfUnits>
        <unit kind="pascal" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="-3" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="Pa_s_per_m4">
      <listOfUnits>
        <unit kind="pascal" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="-4" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="Pa">
      <listOfUnits>
        <unit kind="pascal" exponent="1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m_per_s">
      <listOfUnits>
        <unit kind="metre" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="per_m3">
      <listOfUnits>
        <unit kind="metre" exponent="-3" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="per_m2">
      <listOfUnits>
        <unit kind="metre" exponent="-2" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="mole_per_s">
      <listOfUnits>
        <unit kind="mole" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m2_per_s">
      <listOfUnits>
        <unit kind="metre" exponent="2" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="mole_per_s_per_mM2">
      <listOfUnits>
        <unit kind="mole" exponent="-1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="6" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m3_per_skg">
      <listOfUnits>
        <unit kind="metre" exponent="3" scale="0" multiplier="1"/>
        <unit kind="kilogram" exponent="-1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m2">
      <listOfUnits>
        <unit kind="metre" exponent="2" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="mM_s">
      <listOfUnits>
        <unit kind="mole" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="-3" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="per_s">
      <listOfUnits>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m3">
      <listOfUnits>
        <unit kind="metre" exponent="3" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="mM">
      <listOfUnits>
        <unit kind="mole" exponent="1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="-3" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="mole_per_s_per_mM">
      <listOfUnits>
        <unit kind="metre" exponent="3" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="kg_per_m3">
      <listOfUnits>
        <unit kind="kilogram" exponent="1" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="-3" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="per_mM">
      <listOfUnits>
        <unit kind="metre" exponent="3" scale="0" multiplier="1"/>
        <unit kind="mole" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m6_per_Pa2_s2">
      <listOfUnits>
        <unit kind="pascal" exponent="-2" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-2" scale="0" multiplier="1"/>
        <unit kind="metre" exponent="6" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="kg">
      <listOfUnits>
        <unit kind="kilogram" exponent="1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m">
      <listOfUnits>
        <unit kind="metre" exponent="1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="s">
      <listOfUnits>
        <unit kind="second" exponent="1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="Pa_s">
      <listOfUnits>
        <unit kind="pascal" exponent="1" scale="0" multiplier="1"/>
        <unit kind="second" exponent="1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
    <unitDefinition id="m3_per_s">
      <listOfUnits>
        <unit kind="metre" exponent="3" scale="0" multiplier="1"/>
        <unit kind="second" exponent="-1" scale="0" multiplier="1"/>
      </listOfUnits>
    </unitDefinition>
  </listOfUnitDefinitions>
  <listOfCompartments>
    <compartment id="e" name="external" spatialDimensions="3" units="m3" constant="true"/>
    <compartment id="c" name="cytosol" spatialDimensions="3" units="m3" constant="false"/>
    <compartment id="m" name="membrane" spatialDimensions="2" units="m2" constant="true"/>
  </listOfCompartments>
  <listOfSpecies>
    <species id="c__gal" name="D-galactose" compartment="c" initialConcentration="0.00012" substanceUnits="mole" hasOnlySubstanceUnits="false" boundaryCondition="false" constant="false"/>
    <species id="e__gal" name="D-galactose" compartment="e" initialConcentration="3" substanceUnits="mole" hasOnlySubstanceUnits="false" boundaryCondition="true" constant="false"/>
  </listOfSpecies>
  <listOfParameters>
    <parameter id="x_cell" name="cell diameter" value="2.5e-05" units="m" constant="true"/>
    <parameter id="Vol_e" name="external volume" value="0.0001" units="m3" constant="true"/>
    <parameter id="A_m" name="membrane area" value="1" units="m2" constant="true"/>
    <parameter id="Vol_c" name="cell volume" units="m3" constant="true"/>
    <parameter id="GLUT2_Vmax" value="1e-13" units="mole_per_s" constant="true"/>
    <parameter id="GLUT2_k_gal" value="1" units="mM" constant="true"/>
  </listOfParameters>
  <listOfInitialAssignments>
    <initialAssignment symbol="Vol_c">
      <math xmlns="http://www.w3.org/1998/Math/MathML">
        <apply>
          <times/>
          <ci> x_cell </ci>
          <ci> x_cell </ci>
          <ci> x_cell </ci>
        </apply>
      </math>
    </initialAssignment>
    <initialAssignment symbol="e">
      <math xmlns="http://www.w3.org/1998/Math/MathML">
        <ci> Vol_e </ci>
      </math>
    </initialAssignment>
    <initialAssignment symbol="m">
      <math xmlns="http://www.w3.org/1998/Math/MathML">
        <ci> A_m </ci>
      </math>
    </initialAssignment>
  </listOfInitialAssignments>
  <listOfRules>
    <assignmentRule variable="c">
      <math xmlns="http://www.w3.org/1998/Math/MathML">
        <ci> Vol_c </ci>
      </math>
    </assignmentRule>
  </listOfRules>
  <listOfReactions>
    <reaction id="e__GLUT2_GAL" name="galactose transport [e__]" reversible="true" fast="false" compartment="m">
      <listOfReactants>
        <speciesReference species="e__gal" stoichiometry="1" constant="true"/>
      </listOfReactants>
      <listOfProducts>
        <speciesReference species="c__gal" stoichiometry="1" constant="true"/>
      </listOfProducts>
      <kineticLaw>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:sbml="http://www.sbml.org/sbml/level3/version1/core">
          <apply>
            <divide/>
            <apply>
              <times/>
              <ci> GLUT2_Vmax </ci>
              <apply>
                <minus/>
                <cn sbml:units="dimensionless" type="integer"> 1 </cn>
                <apply>
                  <divide/>
                  <ci> c__gal </ci>
                  <ci> e__gal </ci>
                </apply>
              </apply>
            </apply>
            <apply>
              <plus/>
              <cn sbml:units="dimensionless" type="integer"> 1 </cn>
              <apply>
                <divide/>
                <ci> c__gal </ci>
                <ci> GLUT2_k_gal </ci>
              </apply>
              <apply>
                <divide/>
                <ci> e__gal </ci>
                <ci> GLUT2_k_gal </ci>
              </apply>
            </apply>
          </apply>
        </math>
      </kineticLaw>
    </reaction>
  </listOfReactions>
</model>
</sbml>
```

Result:

![test_1_-_2016-01-11_16 22 33](https://cloud.githubusercontent.com/assets/900538/12237592/cb45a91c-b87f-11e5-9139-ef721cd8c892.png)

When setting steps in addition to start and stop time everything is working. roadrunner should take the default value of 50 steps.

**Comment by hsauro** [comment]
This needs to be fixed and ages added to the python regression suite.

Herbert

**Comment by 0u812** [comment]
fixed in 2324abf76a491fb87345e3d8b5a73ae31fef5578
